tgs/src/main/cpp/tgs/FeatureExtraction/GeneticAlgorithm.cpp

```diff
@@ -290,98 +272,44 @@ namespace Tgs
-//           if (mutate < _mutationProb)
-//           {
-//             brother->mutate(_mutationSeverity);
-//             sister->mutate(_mutationSeverity);
-//           }
-//           // make sure at least one fourth of the new generation is unique
-//           if ((int)nextGen.size() < _populationSize / 4)
-//           {
-//             if (_used.find(brother->toString()) == _used.end())
-//             {
-//               nextGen.push_back(brother);
-//               _used.insert(brother->toString());
-//             }
-//             if ((int)nextGen.size() < _populationSize)
-//             {
-//               if (_used.find(sister->toString()) == _used.end())
-//               {
-//                 nextGen.push_back(sister);
-//                 _used.insert(sister->toString());
-//               }
-//             }
-//           }
-//           else
-          {
             nextGen.push_back(brother);
             _used.insert(brother->toString());
             nextGen.push_back(sister);
             _used.insert(sister->toString());
-          }
         }
       }
     }
-    //cout << "                         \r";
     _population = nextGen;
     _updateScores();
-//     // sort in descending order
-//     sort(_population.begin(), _population.end(), CompareGenomes());
-//     nextGen.clear();
-//     string lastStr = _population[0]->toString();
-//     nextGen.push_back(_population[0]);
-//     for (unsigned int i = 1; i < _population.size(); i++)
-//     {
-//       if (_population[i]->toString() != lastStr)
-//       {
-//         nextGen.push_back(_population[i]);
-//         lastStr = _population[i]->toString();
-//       }
-//     }
-//     _population = nextGen;
   }
 
   void GeneticAlgorithm::_updateScores()
   {
-    bool printedProgress = false;
     for (unsigned int i = 0; i < _population.size(); i++)
     {
       boost::shared_ptr<Genome> g = _population[i];
       double score = g->getScore();
       string str = g->toString();
       _used.insert(str);
-//       if (i % 500 == 0 && i > 0)
-//       {
-//         cout << i + 1 << " / " << _population.size() << "                  \r";
-//         cout.flush();
-//         printedProgress = true;
-//       }
-      //cout << "Updating score: " << str << endl << "  ";
       if (_scoreCaching == false || g->isValidScore() == false)
       {
         HashMap<string, double>::const_iterator it = _allScores.find(str);
         if (_scoreCaching == false || it == _allScores.end())
         {
           score = _fitness->calculateFitness(*g);
-          //cout << "Updating... ";
           _allScores[str] = score;
         }
         else
         {
           score = it->second;
         }
-        //cout << score << endl;
         g->setScore(score);
       }
       if (_best == NULL || score > _best->getScore())
       {
         _best = g;
       }
     }
-    if (printedProgress)
-    {
-      cout << "                       \r";
-    }
```

Comment:
This is the actual finding, `printedProgress` is set to `false` and never changed except for in commented out code that hasn't been changed for years.